AEPCore/Sources/core/MobileCore.swift

```diff
@@ -38,6 +39,31 @@ public final class MobileCore: NSObject {
     /// Pending extensions to be registered for legacy support
     static var pendingExtensions = ThreadSafeArray<Extension.Type>(identifier: "com.adobe.pendingExtensions.queue")
 
+    /// Initializes the AEP SDK with the specified `InitOptions`.
+    /// This automatically registers all boundled extensions and sets up lifecycle tracking.
+    /// You can disable automatic lifecycle tracking using `InitOptions`.
+    /// - Parameters:
+    ///   - options: The `InitOptions` used to configure the SDK.
+    ///   - completion: An optional closure triggered once initialization is complete.
+    @available(iOSApplicationExtension, unavailable)
+    @available(tvOSApplicationExtension, unavailable)
+    @objc(initializeWithOptions:completion:)
+    public static func initialize(options: InitOptions, _ completion: (() -> Void)? = nil) {
+        MobileCoreInitializer.shared.initialize(options: options, completion)
```

Comment:
Nit: Given the circular dependency between these two classes, I prefer something like a MobileCore+Initialize extension containing all this logic along with the API, or at least having MobileCore hold a reference to `MobileCoreInitializer` instead of accessing it via `MobileCoreInitializer.shared`. This approach may not scale well when supporting multiple instances.